scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamBackend.scala

```diff
@@ -61,8 +61,15 @@ object BeamPlanner {
           BeamOp.Source(config, src, srcs(src))
         case (IterablePipe(iterable), _) =>
           BeamOp.FromIterable(iterable, kryoCoder)
-        case (wd: WithDescriptionTypedPipe[a], rec) =>
-          rec[a](wd.input)
+        case (wd: WithDescriptionTypedPipe[_], rec) => {
+          val op = rec(wd.input)
+          wd.descriptions match {
```

Comment:
actually, as commented, I think this is probably running the risk of dropping some descriptions.

I would do:
```scala
op.withName(wd.descriptions.map(_._1).mkString(", "))
```
otherwise I think you will wind up with cases where you lose line numbers.

Reply:
Thanks, I'll try this out.